src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs

```diff
@@ -471,6 +489,43 @@ private void CompileNamedType(NamedTypeSymbol containingType)
             }
 
             var members = containingType.GetMembers();
+
+            // We first compile the accessors that contain 'field' keyword to avoid extra bindings for 'field' keyword when compiling other members.
+            for (int memberOrdinal = 0; memberOrdinal < members.Length; memberOrdinal++)
+            {
+                var member = members[memberOrdinal];
+
+                //When a filter is supplied, limit the compilation of members passing the filter.
+                if (!PassesFilter(_filterOpt, member) ||
+                    member is not SourcePropertyAccessorSymbol { ContainsFieldKeyword: true } accessor)
```

Comment:
>ContainsFieldKeyword [](http://example.com/codeflow?start=65&length=20)

The name of this property is too "strong". To my understanding, it indicates whether the accessor might contain a `field` keyword rather than that it definitely contains it. Consider renaming to make this clear. Doing that in a different PR is fine.

Reply:
Added a PROTOTYPE for now